proto/asn1-pdu/asn1_pdu_to_der.cc

```diff
@@ -0,0 +1,159 @@
+#include "asn1_pdu_to_der.h"
+
+namespace asn1_pdu {
+
+// The maximum level of recursion allowed. Values greater than this will just
+// fail.
+static constexpr size_t kRecursionLimit = 200;
+
+// Signal that the recursion limit has been exceeded by a
+// message or value and to abort further processing.
+static bool kErrRecursionLimitReached = false;
+
+uint8_t ASN1PDUToDER::GetVariableIntLen(const uint64_t value,
+                                          const size_t base) {
+  uint8_t base_bits = log2(base);
+  for (uint8_t num_bits = (sizeof(value) - 1) * __CHAR_BIT__;
+       num_bits >= base_bits; num_bits -= base_bits) {
+    if (value >> num_bits) {
+      return ceil((double)num_bits / base_bits) + 1;
```

Comment:
Why the conversion to double here? That raises some concerns. Was this a compiler warning? Was it related to integer truncation? If it is needed for that reason, https://google.github.io/styleguide/cppguide.html#Casting is the guidance here :)

Reply:
@sleevi double was needed because num_bits/base_bits was rounding down to the nearest integer before it got piped into `ceil`.